src/spell.c

```diff
@@ -160,72 +158,105 @@ char ilet;
 
 /* TRUE: book should be destroyed by caller */
 STATIC_OVL boolean
-cursed_book(bp)
-	struct obj *bp;
-{
+cursed_book(struct obj *bp){
 	int lev = objects[bp->otyp].oc_level;
+
+	boolean was_in_use;
+	boolean already_cursed = bp->cursed;
+	int dmg = 0;
+	
+	if(RoSbook == STUDY_WARD) lev -= 1; //The wardings partially protect you from the magic contained within the book.
 	
-	if(RoSbook == STUDY_WARD) lev -= 1; //The wardings paritally protect you from the magic contained within the book.
+	for (int spell = 0; spell < MAXSPELL; spell++)
```

Comment:
It lets you declare `int spell` inside the `for()` loop? I thought the compiler didn't like that.